src/Sarif.FunctionalTests/v2/ConverterTestData/FortifyFpr/FortifyTest.fpr.sarif

```diff
@@ -0,0 +1,377 @@
+                }
+              }
+            }
+          ],
+          "codeFlows": [
+            {
+              "threadFlows": [
+                {
+                  "locations": [
+                    {
+                      "step": 1,
+                      "location": {
+                        "physicalLocation": {
+                          "fileLocation": {
+                            "uri": "ContentFileHandlingController.cs"
+                          },
+                          "region": {
+                            "startLine": 12,
+                            "endLine": 12,
+                            "snippet": {
+                              "text": "        public CMOEViewModel Model { get; set; }\n\n        [Dependency]\n        public ContentFileViewModel ViewModel { get; set; }\n\n        [Dependency]\n        public ICachedDataProvider CachedData { get; set; }\n"
+                            }
+                          }
+                        },
+                        "message": {
+                          "text": "Variable: value"
+                        }
+                      }
+                    },
+                    {
+                      "step": 2,
+                      "location": {
+                        "physicalLocation": {
+                          "fileLocation": {
+                            "uri": "ContentFileViewModel.cs"
+                          },
+                          "region": {
+                            "startLine": 6,
+                            "endLine": 6,
+                            "snippet": {
+                              "text": "\nnamespace MS.IT.HRE.CMOE.Web.Models\n{\n    public class ContentFileViewModel : BaseViewModel\n    {\n        #region Public Properties\n\n"
+                            }
+                          }
+                        },
+                        "message": {
+                          "text": "Class: ContentFileViewModel"
+                        }
+                      }
+                    }
+                  ]
+                }
+              ]
+            }
+          ],
+          "relatedLocations": [
+            {
+              "physicalLocation": {
+                "fileLocation": {
+                  "uri": "ContentFileViewModel.cs"
+                },
+                "region": {
+                  "startLine": 6,
+                  "endLine": 6
+                }
+              }
+            }
+          ]
+        },
+        {
+          "ruleId": "AC21F232-1D82-49B7-9AB1-46FE84CD6424",
+          "message": {
+            "text": "<Content><Paragraph>The method IsDocumentValid() in [DocumentValidator.cs](1) is not reachable from any method outside the class. It is dead code. Dead code is defined as code that is never directly or indirectly executed by a public method.<AltParagraph>This method is not reachable from any method outside the class.</AltParagraph></Paragraph></Content>"
```

Comment:
>Content [](start = 22, length = 7)

Suggestion, parse <Paragraph> and convert to your plaintext message. Use this content and <AltParagraph> in a richtext/markdown version. Obviously, strip all XML elements.

Reply:
Yes, issue #985. I'll do it now.

---
In reply to: [207674968](https://github.com/Microsoft/sarif-sdk/pull/981#discussion_r207674968) [](ancestors = 207674968)